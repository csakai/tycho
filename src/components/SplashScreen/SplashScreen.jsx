import React from 'react';
import './SplashScreen.css';

export default class SplashScreen extends React.Component {
  render() {
    return <div>SPLASH SCREEN</div>;
  }
}
