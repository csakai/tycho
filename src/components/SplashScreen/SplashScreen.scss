$button-height: 50px;
$button-width: 160px;
$button-spacing: 15px;
$button-transition-speed: 0.25s;

.splash-screen {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  display: flex;
  background-color: #fff;
  font-size: 1.5em;
  z-index: 1000;
  background: url('/static/img/bg-stars-large.png');
  transition: opacity $button-transition-speed;
  opacity: 1;

  &--hide {
    opacity: 0;
    pointer-events: none;
    z-index: 0;
  }

  &__hero {
    width: 50vw;
    height: 20vw;
    background-image: url('/static/img/logo-gradient-light.png');
    background-size: contain;
    background-repeat: no-repeat;
    background-position: center center;
  }

  &__content {
    height: $button-height;
  }

  &__loading {
    width: 500px;
    height: 4px;
    background-color: #808080;
  }

  &__loading-bar {
    background-color: #fff;
    height: 4px;
  }

  &__button {
    position: relative;
    width: $button-width;
    height: $button-height;
    margin: 0 $button-spacing;
    perspective: 1000px;
  }
  
  &__button-anchor {
    margin: 0;
    padding: 0;
    width: 100%;
    height: 100%;
    position: absolute;
    transform-style: preserve-3d;
    transform: translateZ(-($button-height / 2));
    transition: transform $button-transition-speed;
    cursor: pointer;

    &::before, &::after {
      justify-content: center;
      align-items: center;
      margin: 0;
      width: $button-width;
      height: $button-height;
      position: absolute;
      border: 5px solid black;
      box-sizing: border-box;
      content: attr(title);
      display: flex;
    }

    &::before {
      background-color: #fff;
      border-color: #000;
      color: #000;
      transform: rotateY(0deg) translateZ($button-height / 2);
    }

    &::after {
      background-color: #000;
      color: #fff;
      transform: rotateX(90deg) translateZ($button-height / 2);
    }

    &:hover {
      transform: translateZ($button-height / 2) rotateX(-90deg);
    }
  }
}

