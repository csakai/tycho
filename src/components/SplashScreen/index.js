import './SplashScreen.css';

export {default} from './SplashScreen';
