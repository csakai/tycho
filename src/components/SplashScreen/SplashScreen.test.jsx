import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import SplashScreen from './SplashScreen';

describe('SplashScreen Component', () => {
    let component, splashScreen;

    beforeEach(() => {
        component = shallow(<SplashScreen 
            pageText={{}}
            percent={1}
            enterScene={jest.fn()}
        />);

        splashScreen = component.instance();
        splashScreen.props = {
            pageText: {}
        };
    });

    describe('renderEnterButton()', () => {
        it('should render the enter button', () => {
            const result = toJson(splashScreen.renderEnterButton());

            expect(result).toMatchSnapshot();
        });
    });

    describe('renderLoadingBar()', () => {
        it('should render the enter button', () => {
            const result = toJson(splashScreen.renderLoadingBar());

            expect(result).toMatchSnapshot();
        });
    });

    describe('renderUserPrompt()', () => {
        it('should render the enter button if percent is 100', () => {
            splashScreen.props.percent = 100;
            const result = toJson(splashScreen.renderUserPrompt());

            expect(result).toMatchSnapshot();
        });

        it('should render the loading bar if percent is not 100', () => {
            splashScreen.props.percent = 50;
            const result = toJson(splashScreen.renderUserPrompt());

            expect(result).toMatchSnapshot();
        });
    });

    describe('render()', () => {
        it('should render the Splash Screen successfully', () => {
            expect(toJson(component)).toMatchSnapshot();
        });
    });
});
