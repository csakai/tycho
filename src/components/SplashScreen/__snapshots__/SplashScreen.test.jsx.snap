// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen Component render() should render the Splash Screen successfully 1`] = `
<div
  className="splash-screen splash-screen--hide"
>
  <div
    className="splash-screen__hero"
  />
  <div
    className="splash-screen__content"
  >
    <div
      className="splash-screen__loading"
    >
      <div
        className="splash-screen__loading-bar"
        style={
          Object {
            "width": "1%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`SplashScreen Component renderEnterButton() should render the enter button 1`] = `undefined`;

exports[`SplashScreen Component renderLoadingBar() should render the enter button 1`] = `undefined`;

exports[`SplashScreen Component renderUserPrompt() should render the enter button if percent is 100 1`] = `undefined`;

exports[`SplashScreen Component renderUserPrompt() should render the loading bar if percent is not 100 1`] = `undefined`;
